src/optimizer/build_probe_side_optimizer.cpp

```diff
@@ -64,6 +64,37 @@ static inline idx_t ComputeOverlappingBindings(const vector<ColumnBinding> &hays
 	return result;
 }
 
+BuildSize BuildProbeSideOptimizer::GetBuildSizes(LogicalOperator &op) {
+	BuildSize ret;
+	switch (op.type) {
+	case LogicalOperatorType::LOGICAL_COMPARISON_JOIN:
+	case LogicalOperatorType::LOGICAL_CROSS_PRODUCT:
+	case LogicalOperatorType::LOGICAL_ANY_JOIN:
+	case LogicalOperatorType::LOGICAL_DELIM_JOIN: {
+
+		auto &left_child = op.children[0];
+		auto &right_child = op.children[1];
+
+		// resolve operator types to determine how big the build side is going to be
+		op.ResolveOperatorTypes();
+		auto left_tuple_layout = TupleDataLayout();
+		left_tuple_layout.Initialize(left_child->types);
```

Comment:
Just for completeness, could we initialize the `TupleDataLayout` like this?
```c++
auto left_tuple_layout = TupleDataLayout();
auto left_types = left_child->types;
left_types.push_back(LogicalType::HASH);
left_tuple_layout.Initialize(left_types);
```
To also take into account the hash (or next ptr) column that is present in the hash table?

Reply:
I thought about that too, but since we are comparing the cost of building on the right side vs building on the left side, this cost of the hash (or nxt ptr) should be added to the other side as well, so decided to leave it out

Reply:
I think leaving it out might skew the cost a bit, as it then seems like one tinyint column is 16x smaller than one hugeint column, even though they both have the hash column, and in reality the width difference is only around 2-3x.